### PR TITLE
Add READMEs to more crates and CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by sdleffler**

### CODE REVIEW

**Comment by kwf** [comment]
Fantastic expository writing!

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+#![warn(missing_copy_implementations, missing_debug_implementations)]
+#![warn(unused_qualifications, unused_results)]
+#![warn(future_incompatible)]
+#![warn(unused)]
+// Documentation configuration
+#![forbid(broken_intra_doc_links)]
+```
+
+Note that warnings are denied during our CI process, so if you are missing documentation or such,
+your PR *will* fail CI. We also use `clippy` in our CI and disallow warnings there as well.
```
Comment (kwf) [inline CONTRIBUTING.md:60]: Can we add "Thank you again for contributing, we really appreciate it!" here? You know, compliment sandwich style. Would also be good to note that they can feel free to get in touch with us via our (work) emails (or just mine, if you prefer) if they have any questions.

File: dialectic-macro/README.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
+This crate contains the `Session!`, `Transmitter`, and `Receiver` macros for use in Dialectic, as
```
Comment (kwf) [inline dialectic-macro/README.md:1]: Can we link "Dialectic" to the crates.io page for it?

File: dialectic-null/README.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+This crate contains the "null" backend for Dialectic. If you are a user, you will likely never have
```
Comment (kwf) [inline dialectic-null/README.md:1]: Can we link "Dialectic" to the crates.io page for it?

File: dialectic-tokio-mpsc/README.md [hunk 4]
```diff
@@ -0,0 +1,14 @@
+This crate contains the Tokio/MPSC backend for Dialectic. It supports send/receive operations over
```
Comment (kwf) [inline dialectic-tokio-mpsc/README.md:1]: Can we link "Dialectic" to the crates.io page for it?

File: dialectic-tokio-serde/README.md [hunk 5]
```diff
@@ -0,0 +1,21 @@
+The `dialectic-tokio-serde` crate provides Dialectic `Transmitter` and `Receiver` types which are
```
Comment (kwf) [inline dialectic-tokio-serde/README.md:1]: Can we link "Dialectic" to the crates.io page for it?

File: dialectic-tokio-serde-bincode/README.md [hunk 6]
```diff
@@ -0,0 +1,3 @@
+A Tokio-compatible backend which can be used with any transport that supports Tokio's
```
Comment (kwf) [inline dialectic-tokio-serde-bincode/README.md:1]: Can we mention that this is a backend is a plugin for the `dialectic-tokio-serde` crate, which itself is a backend builder for `dialectic` (and link both to crates.io)?

File: dialectic-tokio-serde-json/README.md [hunk 7]
```diff
@@ -0,0 +1,3 @@
+A Tokio-compatible backend which can be used with any transport that supports Tokio's
```
Comment (kwf) [inline dialectic-tokio-serde-json/README.md:1]: Can we mention that this is a backend is a plugin for the `dialectic-tokio-serde` crate, which itself is a backend builder for `dialectic` (and link both to crates.io)?

File: dialectic-tokio-serde/README.md [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+- A `dialectic_tokio_serde::Serializer`/`Deserializer` for converting the sent/received types
+  to/from the wire format. *PLEASE NOTE! These are not the serde `Serializer`/`Deserializer` traits
+  but rather similar traits which also define the possible output/input types of the
+  `Serializer`/`Deserializer.`*
+
+## Tokio/serde backends for Dialectic
+
+Currently, two formats are implemented as sister crates:
+
+- The `dialectic-tokio-serde-bincode` crate, which when provided an `AsyncWrite`/`AsyncRead`
```
Comment (kwf) [inline dialectic-tokio-serde/README.md:17]: Can we link these to their pages on crates.io?

**Review by kwf — CHANGES_REQUESTED** [review]
Really great work. Just a few tweaks, noted in the review! :)

File: CONTRIBUTING.md (hunk 1 above)
Comment (sdleffler) [reply CONTRIBUTING.md:60]: My email's public on my GitHub so I'm fine putting it there :P

**Review by kwf — APPROVED** [review]

**Pull request merged**